arrays-carlos-paula.js

```diff
@@ -0,0 +1,37 @@
+var groceryList = ['chip', 'dip','cookies']
+
+// Write the code that will add "soda" to the end of the original array.
+
+console.log(groceryList.push('soda'))
+console.log(groceryList)
+
+// Write the code that will add "granola" to the end of array without altering the original array.
+
+console.log(groceryList.concat('granola'))
+console.log(groceryList)
+
+// Write the code that will return a subset of the array containing only "chips" and "dip".
+
+console.log(groceryList.slice(0,2))
+
+var subsetList = groceryList.slice(0,2)
+console.log(subsetList)
+
+
+// Write the code that will add "beans" to the "chips" and "dip" array.
+
+console.log(subsetList.concat('Beans'))
+
+// Write the code that will add the number 0 to the beginning of the array.
+var numbers = [2, 4, 6 , 8, 10]
+console.log(numbers.unshift(0))
+// Write the code that will add the number 12 to the end of the array.
+console.log(numbers.push(12))
+console.log(numbers)
+// Write the code that will remove the first number from the array.
+console.log(numbers.shift())
+console.log(numbers)
+// Write the code that will add the number 0 to the beginning of the array without altering the original array. HINT: it's not .unshift You'll have to get creative! ;)
+var zero = 0
+var subset = (numbers.concat(zero))
+console.log(subset.reverse())
```

Comment:
I can appreciate this approach. However to meet the expected output for this question you  just need to refactor some.
console.log(subset) shows an output of `[2, 4, 6, 8, 10, 12, 0]` and reversing it gives you `[0, 12, 10, 8, 6, 4, 2]`
However, the expected output is `[0, 2, 4, 6, 8, 10, 12]`, where the only change to the original array is that zero is in the beginning.
So refactor your approach, 

1. maintain the zero variable
2. assign a variable that will reverse the current array
3. then assign a variable that will concat zero to the new array
4. refactor your console.log with the name of that new array

So it should look similar to the following:
var zero = 0
var subset = numbers.reverse()
var subset2 = (subset.concat(zero))
console.log(subset2.reverse())